File: domain/entities/user/credentials.py
from typing import Optional
from domain.entities.base import ApplicationModel
from domain.entities.user.user import User


class WebCredentials(ApplicationModel):
    """
    Модель пользователя из Web.

    Связывает пользователя: его id и учетные данные.
    """

    user_id: int
    email: str
    password: str


class TelegramCredentials(ApplicationModel):
    """
    Модель пользователя из Telegram.

    Связывает пользователя: его id и чат в Telegram.
    `telegram_id` и `chat_id` являются уникальными и предоставляются Telegram API.
    """

    user_id: int
    telegram_id: int
    chat_id: int


class UserCredentials(ApplicationModel):
    telegram: Optional[TelegramCredentials] = None
    web: Optional[WebCredentials] = None


class UserWithCredentials(ApplicationModel):
    user: User
    credentials: UserCredentials

File: nginx/frontend/js/entities/Credentials.js
export class Credentials {
    constructor({ web, telegram }) {
        this.web = web;
        this.telegram = telegram;
    }
}

export class WebCredentials {
    constructor({ email, password }) {
        this.email = email;
        this.password = password;
    }
}

export class TelegramCredentials {
    constructor({ telegram_id, chat_id }) {
        this.telegram_id = telegram_id;
        this.chat_id = chat_id;
    }
}
